Cargo.toml

```
@@ -38,7 +38,7 @@ package.helix-term.opt-level = 2
 
 [workspace.dependencies]
 tree-sitter = { version = "0.20", git = "https://github.com/helix-editor/tree-sitter", rev = "660481dbf71413eba5a928b0b0ab8da50c1109e0" }
-nucleo = "0.2.0"
+nucleo = "0.4.0"
```

```suggestion
nucleo = "0.4.1"
```

I fixed some crashes that are triggered when using this PR (interactive global search on a large directory)